test/integration/util/util.go

```diff
@@ -177,28 +186,47 @@ func PodDeleted(c clientset.Interface, podNamespace, podName string) wait.Condit
 	}
 }
 
+// PodsCleanedUp returns true if all pods are deleted in the specific namespace.
+func PodsCleanedUp(c clientset.Interface, namespace string) wait.ConditionFunc {
+	return func() (bool, error) {
+		list, err := c.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+		return len(list.Items) == 0, nil
+	}
+}
+
 // SyncInformerFactory starts informer and waits for caches to be synced
 func SyncInformerFactory(testCtx *TestContext) {
-	testCtx.InformerFactory.Start(testCtx.Ctx.Done())
+	testCtx.InformerFactory.Start(testCtx.SchedulerCtx.Done())
 	if testCtx.DynInformerFactory != nil {
-		testCtx.DynInformerFactory.Start(testCtx.Ctx.Done())
+		testCtx.DynInformerFactory.Start(testCtx.SchedulerCtx.Done())
 	}
-	testCtx.InformerFactory.WaitForCacheSync(testCtx.Ctx.Done())
+	testCtx.InformerFactory.WaitForCacheSync(testCtx.SchedulerCtx.Done())
 	if testCtx.DynInformerFactory != nil {
-		testCtx.DynInformerFactory.WaitForCacheSync(testCtx.Ctx.Done())
+		testCtx.DynInformerFactory.WaitForCacheSync(testCtx.SchedulerCtx.Done())
 	}
 }
 
 // CleanupTest cleans related resources which were created during integration test
 func CleanupTest(t *testing.T, testCtx *TestContext) {
 	// Kill the scheduler.
```

Comment:
It's right, but not straightforward. The context is the top one in initializing the test environment so it will definitely kill the scheduler. Let me polish this comment.